Графика для всех страниц

AnnaKhVl opened the pull request:
Первый набросок разметки у главной страницы index.html и страницы каталога catalog.html

---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/35/user/2186809/tasks/5)

---
:mortar_board: [Графика для всех страниц](https://up.htmlacademy.ru/htmlcss/35/user/2186809/tasks/7)

AnnaKhVl committed:
Добавлена навигация в шапке главной страницы. (Добавлен файл с вопросами)

AnnaKhVl committed:
Добавлен main

1. Расписано основное содержимое главной страницы в <main>.
2. Добавлены и закомментированы иконки и контентные фотографии.

AnnaKhVl committed:
Добавлен footer

1. Секция подписки перенесена в футер.
2. Завершена вёрстка index.html

AnnaKhVl committed:
Добавлена страница каталога

1. Добавлена страница каталога.
2. В index добавлено количество в избранном.
Добавлен navigation-link-current в шапку.
Заметка: дописать плитки <div> у Преимуществ.
3. В каталоге не написана форма цены с ползунком: добавить потом.

AnnaKhVl committed:
Added </ul> on 275 line

The <ul> tag wasn't closed - mistake was fixed.

AnnaKhVl committed:
Добавлен <div> в каталог

1. Добавлен <div> в advantages-list
2. Changed hotel stars from <li> to <span>
3. Some class attributes fixes

AnnaKhVl committed:
Merge branch 'master' of https://github.com/AnnaKhVl/2186809-sedona-35

AnnaKhVl committed:
Update notes-questions.txt

AnnaKhVl committed:
Добавлена графика

Добавлена графика (картинки, лого, иконки) на главную страницу и на страницу каталога

pavel-niukalo commented on images/icons/rating-stars-2.svg:
```diff
@@ -0,0 +1 @@
+<svg width="140" height="37" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#a)"><path d="m9 10 2.1 6.5H18l-5.6 4 2.2 6.5L9 23l-5.6 4 2.2-6.5-5.6-4h6.9L9 10Z" fill="#83B3D3"/></g><g clip-path="url(#b)"><path d="m33 10 2.1 6.5H42l-5.6 4 2.2 6.5-5.6-4-5.6 4 2.2-6.5-5.6-4h6.9L33 10Z" fill="#83B3D3"/></g><defs><clipPath id="a"><path fill="#fff" transform="translate(0 10)" d="M0 0h18 ...
```
Здесь лучше экспортировать одну звездочку и потом либо фоном вставить и продублировать (`background-repeat`), либо в разметку будем вставлять необходимое кол-во звезд.

Если понадобится добавить/убавить звезд проще будет `html/css` поправить, чем новую картинку экспортировать

AnnaKhVl replied:
Изменила `img` на: 
 `<svg class="icon-star" width="18" height="17" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="m9 0 2.1 6.5H18l-5.6 4 2.2 6.5L9 13l-5.6 4 2.2-6.5-5.6-4h6.9L9 0Z" fill="#83B3D3"/></svg>`
`<!-- TODO Фоном вставить и продублировать (background-repeat), либо в разметку необходимое кол-во звезд) -->`

Во всех четырех карточках отелей.

pavel-niukalo commented on images/icons/rating-stars-4.svg:
```diff
@@ -0,0 +1 @@
+<svg width="140" height="37" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#a)"><path d="m9 10 2.1 6.5H18l-5.6 4 2.2 6.5L9 23l-5.6 4 2.2-6.5-5.6-4h6.9L9 10Z" fill="#83B3D3"/></g><g clip-path="url(#b)"><path d="m33 10 2.1 6.5H42l-5.6 4 2.2 6.5-5.6-4-5.6 4 2.2-6.5-5.6-4h6.9L33 10Z" fill="#83B3D3"/></g><g clip-path="url(#c)"><path d="m57 10 2.1 6.5H66l-5.6 4 2.2 6.5-5.6-4-5.6 4 2. ...
```
А остальные картинки со звездами удалить соответственно

AnnaKhVl replied:
Поменяла

pavel-niukalo commented on images/icons/search-view-table.svg:
```diff
@@ -0,0 +1 @@
+<svg width="48" height="48" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#a)" fill="#000"><path d="M20 17h-4v6h4v-6ZM32 17h-4v6h4v-6ZM26 17h-4v6h4v-6ZM20 25h-4v6h4v-6ZM32 25h-4v6h4v-6ZM26 25h-4v6h4v-6Z"/></g><rect x="1" y="1" width="46" height="46" rx="3" stroke="#000" stroke-width="2"/><defs><clipPath id="a"><path fill="#fff" transform="translate(16 17)" d="M0 0h16v14H0z"/></ ...
```
Сейчас у нас картинка экспортирована с обводкой

![image](https://user-images.githubusercontent.com/52506235/198205254-8dcd4070-362e-411e-a212-5023f412d096.png)

Но если глянуть на стайлгайд, то обводка должна меняться при фокусе например

![image](https://user-images.githubusercontent.com/52506235/198211423-6e1533bc-1cba-49b2-b452-2a5626cfaf5e.png)

Эту обводку будем делать через `css`, поэтому лучше экспортировать маленькие картинки снизу

AnnaKhVl replied:
Заменила картинки на изображения без обводки

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+    <header class="page-header">
+      <nav class="navigation">
+        <a class="navigation-logo" href="index.html">
+          <img class="page-header-logo" src="images\head-logo-sedona.svg" width="140" height="70" alt="Логотип «Город Седона».">
```
Слешы надо сделать в обратную сторону, это надежнее и плюс валидатор ругается на такие 

![image](https://user-images.githubusercontent.com/52506235/198212299-09fc4f77-acbb-4f0b-9701-d50f3f28ab87.png)

AnnaKhVl replied:
Заменила

_Блин, я забыла в валидаторе проверить всё(( ужас у меня там всё красное..._

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+          <li class="navigation-item">
+            <a class="navigation-link" href="#">
+              <span class="visually-hidden">Поиск</span>
+              <img class="icon-search" src="images\icons\navigation-loop.svg" width="20" height="20" alt="Поиск на сайте">
```
Относится ко всем картинкам, плюс надо точки в конце `alt` для всех картинок (у логотипа кстати есть)

AnnaKhVl replied:
Поправила

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+          </li>
+          <li class="navigation-item">
+            <a class="button button-search-hotel" href="#">
+              <span class="visually-hidden">Поиск гостиницы</span>
+              Хочу сюда!
```
Тут можно убрать скрытый текст, поскольку скринридер проговорит видимый текст и этого будет достаточно

AnnaKhVl replied:
Ок

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+          </li>
+          <li class="navigation-item">
+            <a class="navigation-link" href="#">
+              <span class="visually-hidden">Избранное</span>
+              <img class="icon-favorite" src="images\icons\navigation-favorite.svg" width="18" height="17" alt="Избранное">
```
Тут будет двойное проговаривание, сначала скрытого текста, потом `alt` картинки.

Можно оставить только картинку, чтобы ее проговаривало, а скрытый текст делать в том случае, если например у нас картинка фоном лежит

AnnaKhVl replied:
Поняла, поправила везде

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+    <main class="main-index">
+      <h1 class="visually-hidden">Город Седона в штате Аризона, США.</h1>
+      <img class="logo-welcome" src="images\welcome-sedona.svg" width="458" height="352" alt="«Добро пожаловать в великолепную Седону».">
+      <img class="welcome-background" src="images\index-welcome-background.jpg" width="1200" height="485" alt="Панорама каньона и гор в Седоне.">
```
Эту предлагаю фоном положить

AnnaKhVl replied:
Ок, когда буду прописывать CSS учту. 

Фоном только фоновую картинку или  лого "добро пожаловать" тоже?

pavel-niukalo replied:
Добро пожаловать оставим в разметке, только горы фоном (и нижнюю полосочку)

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+      <h1 class="visually-hidden">Город Седона в штате Аризона, США.</h1>
+      <img class="logo-welcome" src="images\welcome-sedona.svg" width="458" height="352" alt="«Добро пожаловать в великолепную Седону».">
+      <img class="welcome-background" src="images\index-welcome-background.jpg" width="1200" height="485" alt="Панорама каньона и гор в Седоне.">
+        <img class="welcome-background-divider" src="images\welcome-background-divider.svg" width="1200" height="57" alt="">
```
И эту тоже, сделаем через множественный фон

AnnaKhVl replied:
Ок, сделаю комментарий в файле

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+      <section class="advantages">
+        <h2 class="visually-hidden">Преимущества</h2>
+        <ul class="advantages-list">
+          <li class="advantages-item">
+            <div class="advantages-item-block-city">
+              <h3 class="advantages-item-title">Настоящий городок</h3>
+              <p class="advantages-item-description">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+              <img class="advantages-photo photo-one" src="images\photo-1.jpg" width="800" height="384" alt="Старые магазинчики с деревьями перед фасадом и открытой верандой на втором этаже, на фоне знаменитых красных гор Седоны.">
+            </div>
+          </li>
```
Для более [простой стилизации](https://codesandbox.io/s/sedona-pymoy7) картинку давай вынесем за `div`:
```html
<li class="advantages-item">
  <div class="advantages-item-block-city">
    <h3 class="advantages-item-title">Настоящий городок</h3>
    <p class="advantages-item-description">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
  </div>
  
  <img class="advantages-photo photo-one" src="images\photo-1.jpg" width="800" height="384" alt="Старые магазинчики с деревьями перед фасадом и открытой верандой на втором этаже, на фоне знаменитых красных гор Седоны.">
</li>
```

AnnaKhVl replied:
Сделала

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+            <h3 class="advantages-item-title">Мало туристов</h3>
+            <p class="advantages-item-description">Большинство едет в Гранд Каньон и толпится там</p>
+          </li>
+          <div class="advantages-item">
+            <div class="advantages-item-block-bridge"></div>
+              <h3 class="advantages-item-title">Там есть мост дьявола</h3>
+              <p class="advantages-item-description">Да, по нему можно пройти! Если вы осмелитесь, разумеется</p>
+              <img class="advantages-photo photo-two" src="images\photo-2.jpg" width="800" height="384" alt="«Мост Дьявола» в каньоне красных скал и сосновых лесов.">
+            </div>
+          </li>
```
На 82 строке вместо `<div>` надо `<li>` указать

AnnaKhVl replied:
Исправила

А здесь не надо картинку выносить за пределы `div`?

pavel-niukalo replied:
Тоже надо, чтобы опять же проще стилизовать

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+          <li class="advantages-item">
+            <h3 class="advantages-item-title">Жильё</h3>
+            <p class="advantages-item-description">Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
+            <img class="advantages-icon-hotel" src="images\icons\icon-advantages-house.svg" width="76" height="72" type="icon">
```
Эти иконки предлагаю фоном потом сделать

AnnaKhVl replied:
Ок, закомментила в файле

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+        <h2 class="search-hotel visually-hidden">Поиск гостиницы</h2>
+        <p class="search-hotel-title">Заинтересовались?</p>
+        <p class="search-hotel-text">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+        <a class="button button-search-hotel" href="#">ПОИСК ГОСТИНИЦЫ В Седоне</a>
```
Текст можно писать в обычном стиле (первая заглавная, дальше в нижнем регистре), а уже сделать их заглавными мы сможем через `css`

AnnaKhVl replied:
Ок, просто скопировала с макета, а там такая вакханалия. Исправила.

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+      <section class="subscription page-footer main-index">
+        <h2 class="subscription-title">Подпишитесь на рассылку</h2>
+        <p class="subscription-text">Только полезная информация и никакого спама, честное бойскаутское!</p>
+        <img class="subscribe-background" src="images\index-subscribe-background.jpg" width="1200" height="415" alt="Панорама каньона и гор в Седоне.">
```
Это тоже фоном сделаем

AnnaKhVl replied:
Отметила

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+
+      <section class="footer-social">
+        <h2 class="page-footer-title">Соцсети</h2>
+        <ul class="footer-social-list">
+            <li class="footer-social-item">
```
Немного отступы поехали, у нас вложенные элементы должны быть отделены двумя пробелами, тут 4 получилось

AnnaKhVl replied:
Поправила

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+        <h2 class="page-footer-title">Соцсети</h2>
+        <ul class="footer-social-list">
+            <li class="footer-social-item">
+              <a class="button" href="#">
+                <span class="visually-hidden">ВКонтакте</span>
+                <img class="footer-social-icon" src="images\icons\vk.svg" width="25" height="15" type="icon">
+              </a>
```
1. Иконку вставим через `<svg>`, поскольку она меняет цвет при наведении
2. У `img` нет атрибута `type`, но это решится, удалив `img` отсюда и заменив его на `svg`

AnnaKhVl replied:
Вставила `<svg></svg>`

Ещё я видела варианты, что в соцсетях прописываются ссылки, а `.svg` вставляется через `background-image` CSS, мы так не делаем?

pavel-niukalo replied:
Нам нужно менять цвет у иконки при наведении и лучший вариант - вставить через `svg`. Можно и фоновой картинкой конечно, но тогда при наведении придется менять ее, а это лишние запросы на сервер

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+              </a>
+            </li>
+            <li class="footer-social-item">
+              <a class="button" href="#">
```
Соц. сетям надо дать ссылки по тз

![image](https://user-images.githubusercontent.com/52506235/198214427-78736d36-7b56-4b80-a8de-ed74efe4bb4d.png)

AnnaKhVl replied:
Прописала

pavel-niukalo commented on index.html:
```diff
@@ -5,8 +5,161 @@
+        </section>
+        <section class="footer-developer">
+          <h2 class="visually-hidden">Сайт разработан «HTML Academy»</h2>
+          <a class="htmlacademy-link" href="https://htmlacademy.ru/">
+            <img class="htmlacademy-logo" src="images\icons\logo-html-academy.svg" width="115" height="34" alt="Логотип HTML Academy">
+          </a>
```
Тут так же вставим через `svg`, поскольку меняется цвет 

![image](https://user-images.githubusercontent.com/52506235/198214593-e22096ba-7ee1-4e84-911a-e266e6bafdf9.png)

AnnaKhVl replied:
Исправила на `svg`

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+            <a class="navigation-link" href="about.html">О Седоне</a>
+          </li>
+          <li class="navigation-item">
+            <a class="navigation-link navigation-link-current" href="catalog.html">Гостиницы</a>
```
Если мы находимся на данной странице, то ссылку на эту же страницу можно убрать (оставить `href="#"`)

AnnaKhVl replied:
Поменяла.
_Я как пользователь обычно наоборот тыкаю на эту ссылку, чтобы обновить её)_

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+
+        <ul class="navigation-list navigation-user">
+          <li class="navigation-item">
+            <a class="navigation-link" href="#">
+              <span class="visually-hidden">Поиск</span>
+              <img class="icon-search" src="images\icons\navigation-loop.svg" width="20" height="20" alt="Поиск на сайте">
+            </a>
```
Здесь те же замечания, что и в `index`, так что не забудь тут поправить потом тоже

AnnaKhVl replied:
Поправила

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+
+      <header class="inner-header">
+        <h1 class="inner-header-title">Гостиницы Седоны</h1>
+        <img class="filter-background" src="images\catalog-filter-background.png" width="1200" height="408" alt="Панорама каньона и гор в Седоне.">
```
Аналогично фоном

AnnaKhVl replied:
Прописала комментарий в файле. 
Картинку пока оставила, когда пропишу в CSS, то уберу её. Или сейчас убрать?

pavel-niukalo replied:
Как удобнее, главное не забыть удалить

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+        <ul class="breadcrumbs">
+          <li class="breadcrumbs-item">
+            <a class="breadcrumbs-link" href="index.html">
+              <span class="visually-hidden">Главная</span>
+              <img class="breadcrumbs-icon-home" src="images\icons\breadcrumbs-home.svg" width="13" height="13" alt="Главная страница">
+            </a>
+          </li>
+          <li class="breadcrumbs-item">
+            <img class="breadcrumbs-icon-arrow" src="images\icons\breadcrumbs-arrow.svg" width="8" height="11" alt="Стрелка-разделитель">
```
Декоративные картинки можно будет фоном положить

AnnaKhVl replied:
Прописала комментарий в файле

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+            <!-- Прописать как выбранную на макете (Заново скачать иконки без рамок, когда рамки пропишутся в CSS)-->
+          <span class="visually-hidden">Показать таблицу</span>
+        </a>
+        <a class="button layout-button" href="#">
+          <img class="icon-slydeshow" src="images\icons\search-view-slydeshow.svg" width="48" height="48" alt="Слайдшоу">
+          <span class="visually-hidden">Показать слайдшоу</span>
+        </a>
+        <a class="button layout-button" href="#">
+          <img class="icon-list" src="images\icons\search-view-list.svg" width="48" height="48" alt="Список">
+          <span class="visually-hidden">Показать список</span>
+        </a>
```
Тут аналогично, оставить один вариант, либо с картинкой и написать в `alt` "Показать ...", либо картинки фоном и оставляем скрытый текст

AnnaKhVl replied:
Оставлю тогда текст, а картинку фоном в CSS.

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+            <a class="product-card-button button more-information" href="#">Подробнее</a>
+            <a class="product-card-button button add-to-favorite" href="#">В избранное</a>
+            <div class="product-card-rating">
+              <span class="product-card-rating-star four-stars visually-hidden">Четыре звезды</span>
+                  <img class="icon-star" src="images\icons\rating-stars-4.svg" width="140" height="37" alt="">
+              </span>
```
Нет открывающего тега для этого спана

![image](https://user-images.githubusercontent.com/52506235/198216093-80df727c-e94a-4e75-a78a-ec63b0fa2065.png)

AnnaKhVl replied:
Исправила

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+              <span class="product-card-rating-star four-stars visually-hidden">Четыре звезды</span>
+                  <img class="icon-star" src="images\icons\rating-stars-4.svg" width="140" height="37" alt="">
+              </span>
+              <a class="product-card-button button rating" href="#">Рейтинг:
+                <span class="product-card-rating-value">8,5</span></a>
```
Здесь цифру можно в спан не оборачивать, а стилизовать ссылку полностью

AnnaKhVl replied:
Убрала `<span>`

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+            <a class="product-card-button button more-information" href="#">Подробнее</a>
+            <a class="product-card-button button add-to-favorite selected" href="#">В избранном</a>
+            <div class="product-card-rating">
+              <span class="product-card-rating-star four-stars visually-hidden">Четыре звезды</span>
+                  <img class="icon-star" src="images\icons\rating-stars-4.svg" width="140" height="37" alt="">
+              </span>
+              <a class="product-card-button button rating" href="#">Рейтинг:
+                <span class="product-card-rating-value">9,2</span></a>
+            </div>
```
Комментарии выше относятся ко всем карточкам

AnnaKhVl replied:
Исправила

pavel-niukalo commented on catalog.html:
```diff
@@ -0,0 +1,304 @@
+        </form>
+      </section>
+
+      <section class="footer-social">
+        <h2 class="page-footer-title">Соцсети</h2>
+        <ul class="footer-social-list">
+            <li class="footer-social-item">
+              <a class="button" href="#">
+                <span class="visually-hidden">ВКонтакте</span>
+                <img class="footer-social-icon" src="images\icons\vk.svg" width="25" height="15" type="icon">
+              </a>
+            </li>
+            <li class="footer-social-item">
+              <a class="button" href="#">
+                <span class="visually-hidden">Telegram</span>
+                <img class="footer-social-icon" src="images\icons\telegram.svg" width="18" height="16" type="icon">
+              </a>
+            </li>
+            <li class="footer-social-item">
+              <a class="button" href="#">
+                <span class="visually-hidden">YouTube</span>
+                <img class="footer-social-icon" src="images\icons\youtube.svg" width="23" height="18" type="icon">
+              </a>
+            </li>
+          </ul>
+        </section>
+        <section class="footer-contacts">
+          <h2 class="visually-hidden">Контакты</h2>
+          <address class="footer-contacts-address">
+            <a class="footer-contacts-address-phone" href="tel:+78128121212">+7 (812) 812-12-12</a>
+          </address>
+        </section>
+        <section class="footer-developer">
+          <h2 class="visually-hidden">Сайт разработан «HTML Academy»</h2>
+          <a class="htmlacademy-link" href="https://htmlacademy.ru/">
+            <img class="htmlacademy-logo" src="images\icons\logo-html-academy.svg" width="115" height="34" alt="Логотип HTML Academy">
+          </a>
+        </section>
+    </footer>
```
Здесь все так же как в `index`

AnnaKhVl replied:
Исправлено

AnnaKhVl committed:
Исправление замечаний

1. Исправлен "/"
2. Закомментены фоновые изображения для последующего переноса в файл CSS
3. Исправлены названия для скринридера (удалены)
4. Добавлены точки в названия "alt"
5. Замена иконок отображения поиска (таблица, список, слайд) и звездочек
6. Добавлены файлы svg вместо img
7. Добавлены работающие ссылки в footer-social
